packages/techdocs-common/src/stages/publish/local.ts

```diff
@@ -89,24 +89,29 @@ export class LocalPublish implements PublisherBase {
     }
 
     return new Promise((resolve, reject) => {
-      fs.copy(directory, publishDir, err => {
+      fs.copy(directory, publishDir, async err => {
```

Comment:
`fs-extra` does not tread the callback as a Promise explicitly, hence why it might be preferred to stay in Promise land even before making the call to `fs.copy`. This can be achieved by defining the `publish` function as `async` and simply `await`ing `fs.copy` as follows:

```js
async publish(foo): bar {
  // sync code
  try {
    await fs.copy(directory, publishDir);
  } catch (error) {
    this.logger.debug('baz');
    throw error;
  }
  
  const baseUrl = await this.discovery.getBaseUrl('techdocs');
  // async stuff
  return { remoteUrl: 'quux' };
}
```

Reply:
For the sake of PR legibility, I tried to keep the diff really small, but this does make things _a lot_ more readable (and you did the review already anyway!), so I'll refactor.